demos/gplates_global/forward.py

```diff
@@ -0,0 +1,370 @@
+    generate_mesh()
+
+    # Load mesh
+    # If initialising start timestepping_history with zero
+    # If restarting, then load the information from CheckpointFile
+    with CheckpointFile("./simulation_states.h5", "r") as f:
+        mesh = f.load_mesh("firedrake_default_extruded")
+        try:
+            timestepping_history = f.get_timestepping_history(
+                mesh, name="Temperature")
+        except RuntimeError:
+            timestepping_history = {"index": [0], "time": [0.0], "delta_t": [1e-9]}
+
+    bottom_id, top_id = "bottom", "top"
+
+    # for accessing the coordinates
+    X = SpatialCoordinate(mesh)
+    r = sqrt(X[0] ** 2 + X[1] ** 2 + X[2] ** 2)
+
+    # Set up function spaces using the bilinear Q2Q1 element pair:
+    V = VectorFunctionSpace(mesh, "CG", 2)  # Velocity function space (vector)
+    W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+    Q = FunctionSpace(mesh, "CG", 2)  # Temperature function space (scalar)
+    Z = MixedFunctionSpace([V, W])  # Mixed function space.
+
+    # Output function space information:
+    log("Number of Velocity DOF:", V.dim())
+    log("Number of Pressure DOF:", W.dim())
+    log("Number of Velocity and Pressure DOF:", V.dim()+W.dim())
+    log("Number of Temperature DOF:", Q.dim())
+
+    # Test functions and functions to hold solutions:
+    v, w = TestFunctions(Z)
+    z = Function(Z)
+    u, p = split(z)
+
+    # Temperature field
+    T = Function(Q, name="Temperature")
+
+    # For computing layer averages
+    T_avg = Function(Q, name='TemperatureProfile')  # Average temperature field
+    T_dev = Function(Q, name='DeviatoricTemperature')  # Deviatoric temperature field
+
+    # non-dimensionalise constants
+    T0 = Constant(0.091)  # Non-dimensional surface temperature
+    Di = Constant(0.5)  # Dissipation number.
+    H_int = Constant(10.0)  # Internal heating
+    Ra = Constant(5.0e6)  # Rayleigh number
+    Di = Constant(0.5)  # Dissipation number.
+    delta_t = Constant(timestepping_history.get("delta_t")[-1])  # Initial time step
+
+    # Initialise temperature field
+    if timestepping_history.get("index")[-1] == 0:
+        T_initialise(T, ((1.0 - (T0*exp(Di) - T0)) * (rmax-r)))
+    else:
+        with CheckpointFile("./simulation_states.h5", mode="r") as f:
+            T.interpolate(
+                f.load_function(
+                    mesh,
+                    name="Temperature",
+                    idx=timestepping_history.get("index")[-1]
+                )
+            )
+
+    # Initiating a plate reconstruction model
+    plate_receonstion_model = pyGplatesConnector(
+        rotation_filenames=[
+            './gplates_files/Zahirovic2022_CombinedRotations_fixed_crossovers.rot'],
+        topology_filenames=[
+            './gplates_files/Zahirovic2022_PlateBoundaries.gpmlz',
+            './gplates_files/Zahirovic2022_ActiveDeformation.gpmlz',
+            './gplates_files/Zahirovic2022_InactiveDeformation.gpmlz'],
```

Comment:
Our of interest, instead of uploading these files to the repo, is there an option to grab them via some sort of wget command at runtime?

Reply:
If they do end up being required for the library, we could do something like xarray and split into a different repository: https://github.com/pydata/xarray-data. However, definitely not subprocessing out to wget at runtime! Use a library like https://pypi.org/project/pooch/ which handles downloading and caching at the Python level.

Reply:
Well we have the legendary g-drift repo for that. For now I am thinking of simply downloading it from Earthbyte website itself, and do a test for it.